src/@commonsku/styles/tables/SelectionTable.tsx

```diff
@@ -1,24 +1,40 @@
-import React, { useCallback, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import VirtualTable, { VirtualTableProps } from './VirtualTable';
-import { Column } from 'react-table';
+import { Column, CellProps } from 'react-table';
 import { LabeledCheckbox } from '../Input';
 
-export type SelectionTableProps = VirtualTableProps & {
-    onSelectRow?: (row?: object, index?: number) => void;
+export type SelectionTableProps<
+    RowType extends Record<string, unknown> & { selected?: boolean },
+    TableProps,
+    TableFooterProps,
+> = VirtualTableProps<RowType, TableProps, TableFooterProps> & {
+    data: RowType[];
+    onSelectRow?: (row: RowType, index: number) => void;
+    onSelectionChange?: (selectedRows: RowType[]) => void;
 }
 
 type SelectionState = 'none' | 'indeterminate' | 'all';
 
-const SelectionTable = (props: SelectionTableProps) => {
-    const { columns, onSelectRow } = props;
+const SelectionTable = <
+    RowType extends Record<string, unknown> & { selected?: boolean },
+    TableProps,
+    TableFooterProps,
+> (props: SelectionTableProps<RowType, TableProps, TableFooterProps>) => {
+    const { columns, onSelectRow, onSelectionChange } = props;
 
-    const [data, setData] = useState(
-        props.data.map(row => ({
+    const [data, setData] = useState<RowType[]>(
+        props.data.map((row: RowType) => ({
             selected: false,
             ...row,
         })
     ));
 
+    useEffect(() => {
```

Comment:
Rather than use `useEffect`, I think that more idiomatic React would just have you call `onSelectionChange()` directly inside `handleSelectHeader()` and `handleSelectRow()`.    https://react.dev/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes